### PR TITLE
chore: add forking option

### PR DESCRIPTION
**Pull request by kittybest**
# Description

Allow to run a forking chain and start from certain block number.

- [x] contracts/hardhat.config.ts: adding `forking` option in the hardhat config
- [x] .env.example: add `FORKING_URL` and `FORKING_BLOCK_NUM`

## Confirmation

- [x] I have read and understand MACI's [contributor guidelines](https://maci.pse.dev/docs/contributing) and [code of conduct](https://maci.pse.dev/docs/contributing/code-of-conduct).
- [x] I have read and understand MACI's [GitHub processes](https://github.com/privacy-scaling-explorations/maci/discussions/847).
- [x] I have read and understand MACI's [testing guide](https://maci.pse.dev/docs/testing).

### CODE REVIEW

File: cli/hardhat.config.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ const config: HardhatUserConfig = {
     localhost: {
       url: process.env.ETH_PROVIDER || DEFAULT_ETH_PROVIDER,
       accounts: [process.env.ETH_SK || DEFAULT_ETH_SK],
-      loggingEnabled: false,
+      loggingEnabled: true,
```
Comment (ctrlc03) [inline cli/hardhat.config.ts:20]: let's maybe keep logging to false, and if someone needs they can change to true?

File: contracts/hardhat.config.ts [hunk 2]
```diff
@@ -38,6 +38,42 @@ const getCommonNetworkConfig = (networkName: ESupportedChains, chainId: number,
...
+    gasPrice: NETWORKS_DEFAULT_GAS[ESupportedChains.Hardhat],
+    chainId: EChainId.Hardhat,
+    accounts: {
+      mnemonic: TEST_MNEMONIC,
+      path: "m/44'/60'/0'/0",
+      initialIndex: 0,
+      count: 20,
+    },
+    gas: DEFAULT_BLOCK_GAS_LIMIT,
+    loggingEnabled: true,
```
Comment (ctrlc03) [inline contracts/hardhat.config.ts:54]: would probably keep this to false here too, and let users change it to true if neede

**Review by ctrlc03 — CHANGES_REQUESTED** [review]
Thanks!

File: contracts/hardhat.config.ts [hunk 3]
```diff
@@ -38,6 +38,42 @@ const getCommonNetworkConfig = (networkName: ESupportedChains, chainId: number,
...
+    gasPrice: NETWORKS_DEFAULT_GAS[ESupportedChains.Hardhat],
+    chainId: EChainId.Hardhat,
+    accounts: {
+      mnemonic: TEST_MNEMONIC,
+      path: "m/44'/60'/0'/0",
+      initialIndex: 0,
+      count: 20,
+    },
+    gas: DEFAULT_BLOCK_GAS_LIMIT,
+    loggingEnabled: !!(process.env.HARDHAT_LOGGING && process.env.HARDHAT_LOGGING === "true"),
```
Comment (0xmad) [inline contracts/hardhat.config.ts:54]:
```suggestion
    loggingEnabled: process.env.HARDHAT_LOGGING === "true",
```

File: contracts/hardhat.config.ts [hunk 4]
```diff
@@ -56,27 +92,7 @@ const config: HardhatUserConfig = {
...
-      loggingEnabled: false,
-      allowUnlimitedContractSize: true,
-      throwOnTransactionFailures: true,
-      throwOnCallFailures: true,
-      mining: {
-        auto: true,
-        interval: 100,
-      },
-    },
+    hardhat: getHardhatConfig(),
```
Comment (0xmad) [inline contracts/hardhat.config.ts:95]: Why do you need this function? Forking can be added without changing this config logic.

```
    hardhat: {
      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
      gasMultiplier: DEFAULT_GAS_MUL,
      gasPrice: NETWORKS_DEFAULT_GAS[ESupportedChains.Hardhat],
      chainId: EChainId.Hardhat,
      accounts: {
        mnemonic: TEST_MNEMONIC,
        path: "m/44'/60'/0'/0",
        initialIndex: 0,
        count: 20,
      },
      gas: DEFAULT_BLOCK_GAS_LIMIT,
      loggingEnabled: false,
      allowUnlimitedContractSize: true,
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      mining: {
        auto: true,
        interval: 100,
      },
      forkings: process.env.FORKING_URL ? { ... } : undefined,
    },
```

Comment (kittybest) [reply contracts/hardhat.config.ts:95]: just think it would be clear? but this way is also good 👍

**Review by ctrlc03 — APPROVED** [review]

**Pull request merged**